vizro-ai/examples/dashboard_ui/app.py

```diff
@@ -35,151 +41,144 @@
+                OffCanvas(id="dashboard-settings", options=["OpenAI"], value="OpenAI"),
                 # Modal(id="modal"),
             ],
         ),
-        Icon(id="open-settings-id"),
     ],
 )
 
-
-dashboard = CustomDashboard(pages=[plot_page])
+dashboard = CustomDashboard(pages=[dashboard_page])
 
 
-# pure dash callbacks
 @callback(
-    [
-        Output("plot-code-markdown", "children", allow_duplicate=True),
-        Output("graph-id", "figure", allow_duplicate=True),
-        Output("text-area-id", "value"),
-        Output("upload-message-id", "children"),
-    ],
-    [Input("on_page_load_action_trigger_vizro_ai_plot_page", "data")],
-    [State("outputs-store-id", "data")],
-    prevent_initial_call="initial_duplicate",
+    Output("dashboard-settings-api-key", "type"),
+    Input("dashboard-settings-api-key-toggle", "value"),
 )
-def update_data(page_data, outputs_data):
-    """Callback for retrieving latest vizro-ai output from dcc store."""
-    if not outputs_data:
-        raise PreventUpdate
+def show_api_key(value):
+    """Callback to show api key."""
+    return "text" if value else "password"
 
-    ai_response = outputs_data["ai_response"]
-    fig = json.loads(outputs_data["figure"])
-    filename = f"File uploaded: '{outputs_data['filename']}'"
-    prompt = outputs_data["prompt"]
 
-    return ai_response, fig, prompt, filename
+@callback(
+    Output("dashboard-settings-api-base", "type"),
+    Input("dashboard-settings-api-base-toggle", "value"),
+)
+def show_api_base(value):
+    """Callback to show api base."""
+    return "text" if value else "password"
 
 
 @callback(
-    Output("settings", "is_open"),
+    Output("dashboard-settings", "is_open"),
     Input("open-settings-id", "n_clicks"),
-    [State("settings", "is_open")],
+    [State("dashboard-settings", "is_open")],
 )
 def open_settings(n_clicks, is_open):
     """Callback for opening and closing offcanvas settings component."""
     return not is_open if n_clicks else is_open
 
 
 @callback(
-    Output("settings-api-key", "type"),
-    Input("settings-api-key-toggle", "value"),
-)
-def show_api_key(value):
-    """Callback to show api key."""
-    return "text" if value else "password"
-
-
-@callback(
-    Output("settings-api-base", "type"),
-    Input("settings-api-base-toggle", "value"),
+    Output("dashboard-code-markdown", "children"),
+    [
+        Input("dashboard-text-area", "value"),
+        Input("dashboard-model-dropdown", "value"),
+        Input("dashboard-settings-api-key", "value"),
+        Input("dashboard-settings-api-base", "value"),
+        Input("dashboard-trigger-button", "n_clicks"),
+        Input("dashboard-data-store", "data"),
+        Input("dashboard-settings-dropdown", "value"),
+    ],
 )
-def show_api_base(value):
-    """Callback to show api base."""
-    return "text" if value else "password"
+def run_script(user_prompt, model, api_key, api_base, n_clicks, data, vendor):  # noqa: PLR0913
```

Comment:
Just run it locally, we probably still need to render something to showing the dashboard generation is in progress. Otherwise it is quite confusing.

One option is to add loading/spinner like when chart UI is running,
Second option probably is more intuitive for dashboard - could we render the progress bar in UI? like when it's running in notebook